graphql/ui_plugin_build_baron.go

```diff
@@ -8,17 +8,113 @@ import (
 	"time"
 
 	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/model/notification"
 	"github.com/evergreen-ci/evergreen/model/task"
 	"github.com/evergreen-ci/evergreen/thirdparty"
+	"github.com/evergreen-ci/evergreen/trigger"
+	"github.com/evergreen-ci/evergreen/units"
+	"github.com/evergreen-ci/utility"
 	"github.com/mitchellh/mapstructure"
 	"github.com/mongodb/grip"
 	"github.com/pkg/errors"
 )
 
 const (
-	jiraSource = "JIRA"
+	jiraSource    = "JIRA"
+	jiraIssueType = "Build Failure"
 )
 
+// bbFileTicket creates a JIRA ticket for a task with the given test failures.
+func BbFileTicket(taskId string, context context.Context) (bool, error) {
```

Comment:
by convention, the context should be the first parameter. https://blog.golang.org/context has some explanation on why ("conclusion" section at the bottom)